src/accessors/IIterator.ts

```diff
@@ -1,6 +1,6 @@
-import { IIteratorResult } from './IIteratorResult';
-
 export interface IIterator<T> {
-    next(skip: number): IIteratorResult<T>;
+    next(packageSize?: number): Array<T>;
     [Symbol.iterator](): IIterator<T>;
+
+    hasMoreElements(): boolean;
```

Comment:
@graywolf336 The iterator anyway has to have a state. Exposing the completeness with a dedicated methods makes it possible to get rid of the `IIteratorResult`, imho